DEVELOP.md

```diff
@@ -0,0 +1,485 @@
+- [Hidden Groups](https://daffodil.apache.org/dev/design-notes/hidden-groups/)
+- [Home](https://daffodil.apache.org/)
+- [Infoset Inputters and Runtime 1 Streaming Unparser & NextElementResolver](https://daffodil.apache.org/dev/design-notes/infoset-inputter-streaming-unparser/)
+- [Interactive Debugger](https://daffodil.apache.org/debugger/)
+- [Namespace Binding Minimization](https://daffodil.apache.org/dev/design-notes/namespace-binding-minimization/)
+- [People](https://daffodil.apache.org/people/)
+- [Releases](https://daffodil.apache.org/releases/)
+- [Runtime2 ToDos](https://daffodil.apache.org/dev/design-notes/runtime2-todos/)
+- [Standard DFDL Schema Project Layout](https://daffodil.apache.org/dfdl-layout/)
+- [Term Sharing in the Schema Compiler](https://daffodil.apache.org/dev/design-notes/term-sharing-in-schema-compiler/)
+- [Test Data Markup Language](https://daffodil.apache.org/tdml/)
+- [Understanding dfdl:bitOrder](https://daffodil.apache.org/tutorials/bitorder.tutorial.tdml.xml)
+- [Unsupported Features and Errata](https://daffodil.apache.org/unsupported/)
+- [VS Code Extension](https://daffodil.apache.org/vscode/)
+- [design notes folder](https://daffodil.apache.org/dev/)
+
+### Tests
+
+The majority of Daffodil tests are written inside TDML files with
+corresponding Scala unit tests which allow your SBT or IDE to run
+these tests within the TDML files.  Daffodil also has some pure Scala
+unit tests for testing smaller parts of Daffodil at a lower level.
+Both kinds of tests are important, although new Daffodil tests tend to
+be written in TDML and focus on testing issues with Daffodil's
+processing of DFDL schemas and parsing data or unparsing data.
+
+TDML tests are helpful for Daffodil developers because they allow a
+developer to write a TDML file with a schema, a test case, and test
+data which isolates a complex data format issue so the developer can
+work on that issue more easily.  The developer is free to decide
+whether to define the schema and test data inline in the TDML file or
+leave the schema and test data in separate files which the TDML file
+reads.
+
+The developer also can choose which TDML implementation Daffodil
+should use to run a TDML test, either Daffodil's Scala backend or
+Daffodil's C code generator, simply by passing an option to Daffodil's
+test command on the command line or passing a TDMLImplementation
+instance to Daffodil's TDML Runner class in a Scala test class.  The
+developer can use a TDML test to work on a part of DFDL which needs to
+be supported by the C code generator and keep running (or debugging)
+the TDML test until the generated C code works successfully:
+
+    daffodil test -I daffodilC nums.tdml
+
+Daffodil also has integration tests in the daffodil-cli module which
+test Daffodil's command line interface.  These tests used to run
+Daffodil in a subprocess which took at least 20 minutes but now almost
+all of them run Daffodil in a thread which takes around 2 minutes.  If
```

Comment:
While this is a great improvement, I'm not sure this sentence really belongs in this document. We don't call out how long it takes to run other tests.

Reply:
OK, I've removed the sentence but added how to run the integration tests.